### ISSUE TITLE
Consider better handling of editing read-only files

### ISSUE THREAD

**Issue by knowledgejunkie** [mu-editor/mu#745]
This bug is forwarded from the Debian Bug report here: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=919754

If you are reporting a bug, we would like to know:

* What you were trying to do,

Open, modify and then save a read-only file

* What steps you took to make this happen,

Open a read-only file on the filesystem (in this case, a MicroPython example script) and make change(s) to it.

* What you expected to happen,

After showing a possible warning about not being able to save the file to its current location (read-only), display a "Save As..." dialog to allow the modified file to be saved elsewhere (e.g. default to ~/mu_code).

* What actually happened,

An endless loop of pop-ups displayed every few seconds showing the Error message:

"Error saving file to disk. Ensure you have permission to write the file and sufficient disk space."

* Why this difference is problematic (it may not be a bug!),

Files may be opened from read-only mountpoints (network shares, directories owned by other users, even optical media, etc) and the user should expect to be able to save modified versions to their local filesystem.

The current behaviour does not allow a modified read-only file to be saved at all. Aside from the timer-triggered Error pop-up loop, the "Save" button also displays the Error message, rather than a "Save As" dialog.

* Technical details like the version of Mu you're using, your OS version and
  other aspects of the context in which Mu was running. 

mu-editor 1.0.2. Debian "Buster".

**Comment by tjguk** [comment]
@petterreinholdtsen that's true. And, it occurs to me, is likely to be especially pertinent for teachers who are looking at / showing students' code on a screen.

The reason I preferred it is because it avoids trying to determine the "right" time to save later on -- autosave, user-save, exit-save. You just save up front and then you're happy.

But your point may well outweigh those advantages
